spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/support/LRUResourceLoader.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.resource.support;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.util.ClassUtils;
+
+/**
+ * A wrapper around a ResourceLoader that limits the space used on disk of resolved resources by
+ * removing the least recently resolved entries.
+ *
+ * @author Eric Bottard
+ */
+public class LRUResourceLoader implements ResourceLoader {
+
+	private static final Logger logger = LoggerFactory.getLogger(LRUResourceLoader.class);
+
+	private final ClassLoader classLoader = ClassUtils.getDefaultClassLoader();
+
+	private final ResourceLoader delegate;
+
+	private final Map<File, Long> cache;
+
+	/**
+	 * Wrap an existing ResourceLoader, limiting total used space to {@literal maxSize} bytes.
+	 */
+	public LRUResourceLoader(ResourceLoader delegate, long maxSize) {
```

Comment:
I don't see how we can know the size of this specific file up front to set it to an appropriate value.  Consider the case where we have 1GB of free disk.  I should be able to have a 500MB .jar file and 10 * 50MB other artfiacts or have 100 * 50 MB file.  How to set the maxSize in this case?